File: web/src/api/storage.ts
import { http } from '@/lib/http.ts';

// get image list
export function getImages() {
  return http.get('/api/storage/image');
}

// get mounted image
export function getMountedImage() {
  return http.get('/api/storage/image/mounted');
}

// mount/unmount image
export function mountImage(file?: string, cdrom?: boolean) {
  const data = {
    imageId: file ? file : '',
    cdrom: cdrom
  };
  return http.request({ method: 'post', url: '/api/storage/image/mount', data });
}

// get CD-ROM flag
export function getCdRom() {
  return http.get('/api/storage/cdrom');
}

export function deleteImage(file: string) {
  const data = {
    imageId: file
  };
  return http.request({ method: 'post', url: '/api/storage/image/delete', data });
}

File: server/router/storage.go
package router

import (
	"github.com/gin-gonic/gin"

	"NanoKVM-Server/middleware"
	"NanoKVM-Server/service/storage"
)

func storageRouter(r *gin.Engine) {
	service := storage.NewService()
	api := r.Group("/api")
	api.Use(middleware.CheckToken())

	api.GET("/storage/image", service.GetImages)               // get image list
	api.GET("/storage/image/mounted", service.GetMountedImage) // get mounted image
	api.GET("/storage/cdrom", service.GetCdRom)                // get CD-ROM flag
	stepUp := api.Group("/storage").Use(middleware.CheckToken())
	stepUp.POST("/image/mount", service.MountImage)
	stepUp.POST("/image/delete", service.DeleteImage)
}
